flow/aggregates.py

```diff
@@ -0,0 +1,463 @@
+        provided by the user then the last aggregate will have the maximum possible
+        jobs in it.
+        For instace, if 10 jobs are present in a project and they are aggregated in groups
+        of 3 then the length of aggregates would be 3, 3, 3, 1.
+
+        The below code-block provides an example on how jobs can be aggregated in
+        groups of 2.
+
+        .. code-block:: python
+
+            @Aggregate.groupsof(num=2)
+            @FlowProject.operation
+            def foo(*jobs):
+                print(len(jobs))
+
+        :param num:
+            Maximum possible length of an aggregate.
+        :type num:
+            int
+        :param sort_by:
+            Before aggregating, sort the jobs by a given statepoint parameter.
+            The default behavior is no sorting.
+        :type sort_by:
+            str or NoneType
+        :param reverse_order:
+            States if the jobs are to be sorted in reverse order.
+            The default value is False.
+        :type reverse_order:
+            bool
+        :param select:
+            Condition for filtering individual jobs. This is passed as the
+            callable argument to `filter`.
+            The default behavior is no filtering.
+        :type select:
+            callable or NoneType
+        """
+        try:
+            if num != int(num):
+                raise ValueError('The num parameter should be an integer')
+            num = int(num)
+            if num <= 0:
+                raise ValueError('The num parameter should have a value greater than 0')
+        except TypeError:
+            raise TypeError('The num parameter should be an integer')
+
+        def aggregator(jobs):
+            args = [iter(jobs)] * num
+            return zip_longest(*args)
+
+        setattr(aggregator, '_num', num)
+
+        return cls(aggregator, sort_by, reverse_order, select, num=num)
+
+    @classmethod
+    def groupby(cls, key, default=None, sort_by=None, reverse_order=False, select=None):
+        """signac jobs can be aggregated by a valid state point parameter.
+
+        Users can aggregate jobs by passing a single or a list of state point parameters
+        or a custom callable.
+        If a user provides a list of state point parameters then the default value, if passed,
+        expects a list of default values having length equal to the length of key passed.
+        By default, an error is raised if the key passed is invalid.
+
+        .. code-block:: python
+
+            @Aggregate.groupby('sp', -1)
+            @FlowProject.operation
+            def foo(*jobs):
+                print(len(jobs))
+
+        :param key:
+            Parameter which specifies how the jobs should be aggregated.
+        :type key:
+            str, Iterable, or callable
+        :param key:
+            Default values used for grouping if invalid key is passed
+        :type key:
+            str, Iterable, or callable
```

Comment:
the key parameter needs more explanation. Also, you have the key parameter listed twice.

Reply:
Oh, I'm sorry about that. Thanks for letting me know.